structuredheadings/plugin.js

```diff
@@ -222,24 +183,162 @@
 
       setupElements(editor);
       setupCommands(editor);
-      addButtons(editor);
       setupStyles(editor);
 
-      //Dialogs
-      //eslint-disable-next-line new-cap
-      editor.addCommand("selectStyle", new CKEDITOR.dialogCommand("selectStyle", {
-        startDisabled: true,
-        contextSensitive: true,
-        refresh: function () {
-          if (isEmpty(editor.config.autonumberStyles)) {
-            this.setState(CKEDITOR.TRISTATE_DISABLED);
+      //Format Dropdown
+      editor.ui.addRichCombo("NumFormats", {
+        label: "Formats",
+        title: "Numbering Formats",
+        toolbar: "styles,7",
+        allowedContent: editor.config.numberedElements.concat(["pre", "p", "div"]),
+
+        panel: {
+          css: [ CKEDITOR.skin.getPath("editor") ].concat(editor.config.contentsCss),
+          multiSelect: false,
+          attributes: { "aria-label": "Numbering Formats" }
+        },
+
+        init: function () {
+
+          this.startGroup("Formats");
+
+          this.add("p",
+            elementStyles.p.buildPreview(
+              "Normal Text"
+            ),
+            "Normal Text"
+          );
+          for (var key in editor.config.numberedElements) {
+            this.add(editor.config.numberedElements[key],
+              elementStyles[ editor.config.numberedElements[key] ].buildPreview(
+                "Header " + editor.config.numberedElements[key].slice(1)
+              ),
+              "Header " + editor.config.numberedElements[key].slice(1));
+          }
+          this.add("pre",
+              elementStyles.pre.buildPreview(
+                  "Formatted Text"
+                ),
+            "Preformatted Text"
+          );
+        },
+
+        onClick: function (value) {
+          editor.applyStyle(elementStyles[ value ]);
+          setLevel(editor, editor.elementPath().block);
+          editor.execCommand("reapplyStyle");
+          if (value === "p") {
+            this.setValue(value, "Normal Text");
+          } else if (value === "pre") {
+            this.setValue(value, "Formatted Text");
           } else {
-            this.setState(CKEDITOR.TRISTATE_OFF);
+            this.setValue(value, "Header " + editor.config.numberedElements[
+                editor.config.numberedElements.indexOf(value)
+              ].slice(1));
```

Comment:
updated, expanded... made sure things are aligned, I'm not sure what correct indentation is, I have the list index split, the closing bracket aligned with the start of the line where it is opened.